File: solutions/erlang/anagram/1/anagram.erl
-module(anagram).
-export([find/2]).

-spec find(string(), [string()]) -> string().
find(Word, List) ->
  Normalized = string:to_lower(Word),
  [W || W <- List, is_anagram(Normalized, string:to_lower(W))].

-spec is_anagram(string(), string()) -> boolean().
is_anagram(Word, Word) ->
  % A word is not considered an anagram of itself.
  false;
is_anagram(Word, Candidate) ->
  lists:sort(Word) =:= lists:sort(Candidate).

File: solutions/java/anagram/1/src/main/java/Anagram.java
import java.util.Arrays;
import java.util.List;
import java.util.function.Predicate;
import java.util.stream.Collectors;

public class Anagram {

    private final Predicate<String> isAnagram;

    public Anagram(String master) {
        Predicate<String> notSameWord = word -> !master.equalsIgnoreCase(word);
        String sorted = anagramSort(master);
        this.isAnagram = notSameWord.and(word -> sorted.equals(anagramSort(word)));
    }

    public List<String> match(List<String> strings) {
        return strings.stream()
                      .filter(isAnagram)
                      .collect(Collectors.toList());
    }

    private static String anagramSort(String word) {
        char[] chars = word.toLowerCase().toCharArray();
        Arrays.sort(chars);
        return String.valueOf(chars);
    }
}